### PR TITLE
(ITE-146) Update Segment Engage Destination to handle onDelete events

### PR DESCRIPTION
**Pull request by illumin04**
Update Segment Engage Destination to handle onDelete events
A logic rewrite after the new graphQL API.

_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [If destination is already live] Tested for backward compatibility of destination. **Note:** New required fields are a breaking change.
- [ ] [Segmenters] Tested in the staging environment
- [ ] [Segmenters] [If applicable for this change] Tested for regression with Hadron.

### CODE REVIEW

File: packages/destination-actions/src/destinations/stackadapt-audiences/index.ts [hunk 1]
```diff
@@ -60,10 +60,55 @@ const destination: DestinationDefinition<Settings> = {
       }
     }
   },
+
+  onDelete: async (request, { payload }) => {
+    const userId = payload.userId
+    const formattedExternalIds = `["${userId}"]`
+
+    const syncIds = [sha256hash(String(userId))]
+    const formattedSyncIds = `[${syncIds.map((syncId: string) => `"${syncId}"`).join(', ')}]`
```
Comment (AlliterativeAlice) [inline packages/destination-actions/src/destinations/stackadapt-audiences/index.ts:69]: If I understand correctly, we know that there will always only be one syncId, so why put it into an array and map it if the outcome will always just be `"${syncId}"`

Comment (illumin04) [reply packages/destination-actions/src/destinations/stackadapt-audiences/index.ts:69]: Yes that's correct, this was leftover logic from before, I will fix it

File: packages/destination-actions/src/destinations/stackadapt-audiences/index.ts [hunk 2]
```diff
@@ -65,14 +65,13 @@ const destination: DestinationDefinition<Settings> = {
...
-    const syncIds = [sha256hash(String(userId))]
-    const formattedSyncIds = `[${syncIds.map((syncId: string) => `"${syncId}"`).join(', ')}]`
+    const syncId = [sha256hash(String(userId))]
 
     const mutation = `mutation {
       deleteProfilesWithExternalIds(
         externalIds: ${formattedExternalIds},
         externalProvider: "${EXTERNAL_PROVIDER}",
-        syncIds: ${formattedSyncIds}
+        syncIds: ${syncId}
```
Comment (AlliterativeAlice) [inline packages/destination-actions/src/destinations/stackadapt-audiences/index.ts:74]: I don't think the brackets are preserved when you convert an array to a string, if you want that

Comment (illumin04) [reply packages/destination-actions/src/destinations/stackadapt-audiences/index.ts]: They should be called like syncId: "28test", so I adjusted the format to that

Comment (AlliterativeAlice) [reply packages/destination-actions/src/destinations/stackadapt-audiences/index.ts]: Okay, but what is the purpose of the syncId variable still being an array?

Comment (illumin04) [reply packages/destination-actions/src/destinations/stackadapt-audiences/index.ts]: Right, I'll fix it as well

**Review by AlliterativeAlice — APPROVED** [review]

**Pull request merged**